master/buildbot/process/build.py

```diff
@@ -284,30 +286,14 @@ def startBuild(self, build_status, workerforbuilder):
             # handler
             log.msg("Build.setupBuild failed")
             log.err(Failure())
-            self.finished = True
-            self.results = EXCEPTION
-            self.deferred = None
+            self.buildFinished(['Build.setupBuild', 'failed'], EXCEPTION)
             return
         # flush properties in the beginning of the build
         yield self._flushProperties(None)
-        yield self.master.data.updates.setBuildStateString(self.buildid,
-                                                           u'starting')
-        self.build_status.buildStarted(self)
```

Comment:
I dont think self.build_status.buildStarted(self) is of any use now.
It used to notify status plugin, but there are none now.